compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -820,6 +823,16 @@ class TreeUnpickler(reader: TastyReader,
         // Child annotations for local classes and enum values are not pickled, so
         // need to be re-established here.
         sym.registerIfChild(late = true)
+
+      if (ctx.mode.is(Mode.ReadComments)) {
+        for { docCtx <- ctx.docCtx
```

Comment:
If the mode `ReadComments`, then I suppose we should both `docCtx` and `commentUnpickler` should not be empty

Reply:
The same goes for `posUnpicklerOpt`: we check if the mode is `ReadPositions` and whether `posUnpicklerOpt` is set.

The `TreeUnpickler` can be constructed with a certain configuration in mind (i.e. not defining `commentUnpicklerOpt`), and then used with a different configuration (i.e. `ctx.mode.is(Mode.ReadComments)`), which would lead to crashes if we just used `.get` on the `Option`.

Do we prefer crashing? Should we add a warning?

Reply:
I really can't tell why the position unpickler is taken as an option.

I don't see a good reason to do the same. It only makes thing more complicated. I think the `TreeUnpickler` does not need an optional `CommentUnpickler`. I think we should just look at `Mode.ReadComments` to decide whether or not we unpickle comments. And in the case we do, we should assert that we have a `docCtx`.

Maybe @smarter or @nicolasstucki can explain why it was done like this for `PositionUnpickler`

Reply:
No idea, but what you said sounds reasonable

Reply:
It's actually valid for the `positionUnpickler` or `commentUnpickler` to be `None` if there are no `Positions` or `Comments` sections in the TASTY file from which we're unpickling.